kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/nulls.kt

```diff
@@ -5,114 +5,446 @@ import io.kotest.matchers.MatcherResult
 import io.kotest.matchers.should
 import io.kotest.matchers.shouldNot
 
+/**
+ * Asserts that this iterable contains only null elements.
+ *
+ * Verifies that all elements in the iterable are null.
+ *
+ * Opposite of [shouldNotContainOnlyNulls].
+ *
+ * Example:
+ * ```
+ * listOf(null, null).shouldContainOnlyNulls() // Assertion passes
+ * listOf(null, "a").shouldContainOnlyNulls()  // Assertion fails
+ * ```
+ *
+ * @return the same iterable for further assertions
+ */
 fun <T, I : Iterable<T>> I.shouldContainOnlyNulls(): I {
    toList().shouldContainOnlyNulls()
    return this
 }
 
+/**
+ * Asserts that this array contains only null elements.
+ *
+ * Verifies that all elements in the array are null.
+ *
+ * Opposite of [shouldNotContainOnlyNulls].
+ *
+ * Example:
+ * ```
+ * arrayOf(null, null).shouldContainOnlyNulls() // Assertion passes
+ * arrayOf(null, "a").shouldContainOnlyNulls()  // Assertion fails
+ * ```
+ *
+ * @return the same array for further assertions
+ */
 fun <T> Array<T>.shouldContainOnlyNulls(): Array<T> {
    asList().shouldContainOnlyNulls()
    return this
 }
 
+/**
+ * Asserts that this collection contains only null elements.
+ *
+ * Verifies that all elements in the collection are null.
+ *
+ * Opposite of [shouldNotContainOnlyNulls].
+ *
+ * Example:
+ * ```
+ * listOf(null, null).shouldContainOnlyNulls() // Assertion passes
+ * listOf(null, "a").shouldContainOnlyNulls()  // Assertion fails
+ * ```
+ *
+ * @return the same collection for further assertions
+ */
 fun <T, C : Collection<T>> C.shouldContainOnlyNulls(): C {
    this should containOnlyNulls()
    return this
 }
 
+/**
+ * Asserts that this iterable does not contain only null elements.
+ *
+ * Verifies that the iterable contains at least one non-null element.
+ *
+ * Opposite of [shouldContainOnlyNulls].
+ *
+ * Example:
+ * ```
+ * listOf(null, "a").shouldNotContainOnlyNulls()  // Assertion passes
+ * listOf(null, null).shouldNotContainOnlyNulls() // Assertion fails
+ * ```
+ *
+ * @return the same iterable for further assertions
```

Comment:
I know this is preexisting, but because Collection inherits from Iterable, why do we need both methods? Should we deprecate the extension method on Collection?

Reply:
I think I asked you this a few months ago 😂 

I have no problem with plain removing it. It won't change user's code